modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/js/ProductOptions.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+// import slugify from 'commerce-frontend-js/utilities/slugify'
+
+import {debounce} from 'frontend-js-web';
+
+
+export default function ({namespace}) {
+	console.log('cucu')
+	var form = document.getElementById('#'+namespace + 'fm');
+
+	var keyInput = form.querySelector('#'+namespace+ 'key');
+	var nameInput = form.querySelector('#'+namespace + 'name');
+
+	var handleOnNameInput = function (event) {
+		keyInput.value = slugify.default(nameInput.value);
+	};
+
+	nameInput.addEventListener('input', debounce.(handleOnNameInput, 200));
```

Comment:
This is a syntax error since `debounce` is a function. This should be just `debounce(...)`. As it is, the build shows the following error:

> { SyntaxError: /Users/jose.balsas/dev/liferay/portal/master/portal/modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/js/ProductOptions.js: Unexpected token (31:46)
>
>  29 | 	};
>  30 | 
>  31 | 	nameInput.addEventListener('input', debounce.(handleOnNameInput, 200));

For some reason, our build finishes successfully, but the files aren't built properly so you're likely to see after this something like:

> Uncaught SyntaxError: Cannot use import statement outside a module

or 

>  A require() call has failed but no failure handler was provided.

These are nothing but red herrings distracting you from the fact that your build failed and produced a bad output. 

Correcting this and re-deploying I see that `debounce` is the expected function.

/cc @bryceosterhaus, @wincent, we might want to look into this. Is it possible we've somehow messed with the exit codes or failed to catch them and send them upwards to gradle? This case apparently throws an unhandled exception which might be falling through the cracks:

> (node:51583) UnhandledPromiseRejectionWarning: Error: Command /Users/jose.balsas/dev/liferay/portal/master/portal/modules/node_modules/@liferay/npm-scripts/node_modules/.bin/babel --no-babelrc --config-file /Users/jose.balsas/dev/liferay/portal/master/portal/modules/apps/commerce/commerce-product-options-web/build/npmscripts/tmp/.babelrc src/main/resources/META-INF/resources --out-dir build/node/packageRunBuild/resources --source-maps exited with code 1

Reply:
Ugh — it must be something to do with the ~Gradle environment~ Yarn version; I only see `UnhandledPromiseRejectionWarning` and the false `BUILD SUCCESSFUL` from Gradle.

When I fetch this PR and `cd` into `modules/apps/commerce/commerce-product-options-web`, I see `yarn build` exit like this (edited for brevity):

```
yarn run v1.22.5
warning package.json: No license field
$ liferay-npm-scripts build
Using NODE_ENV: "production"
{ SyntaxError: /Users/greghurrell/code/portal/liferay-portal/modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/js/ProductOptions.js: Unexpected token (31:46)

  29 |  };
  30 |
> 31 |  nameInput.addEventListener('input', debounce.(handleOnNameInput, 200));
     |                                               ^
  32 |
  33 |  document
  34 |          .getElementById( namespace +'publishButton')
    at Object._raise (/Users/greghurrell/code/portal/liferay-portal/modules/node_modules/@babel/parser/lib/index.js:763:17)
  loc: Position { line: 31, column: 46 },
  pos: 1091,
  code: 'BABEL_PARSE_ERROR' }
Command /Users/greghurrell/code/portal/liferay-portal/modules/node_modules/@liferay/npm-scripts/node_modules/.bin/babel --no-babelrc --config-file /Users/greghurrell/code/portal/liferay-portal/modules/apps/commerce/commerce-product-options-web/build/npmscripts/tmp/.babelrc src/main/resources/META-INF/resources --out-dir build/node/packageRunBuild/resources --source-maps exited with code 1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
zsh: exit 1     yarn build
```

But when I run via Gradle, the error gets printed but the build spuriously claims success (again, edited for brevity):

```
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=2048m; support was removed in 8.0
> Task :yarnInstall
> Task :apps:commerce:commerce-product-options-web:packageRunBuild
yarn run v1.13.0
warning package.json: No license field
$ liferay-npm-scripts build
Using pre-existing NODE_ENV: "development"
{ SyntaxError: /Users/greghurrell/code/portal/liferay-portal/modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/js/ProductOptions.js: Unexpected token (31:46)

  29 |  };
  30 |
> 31 |  nameInput.addEventListener('input', debounce.(handleOnNameInput, 200));
     |                                               ^
  32 |
  33 |  document
  34 |          .getElementById( namespace +'publishButton')
    at Object._raise (/Users/greghurrell/code/portal/liferay-portal/modules/node_modules/@babel/parser/lib/index.js:763:17)
(node:14966) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:14966) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Done in 4.94s.

> Task :apps:commerce:commerce-product-options-web:processResources
> Task :apps:commerce:commerce-product-options-web:deployConfigs NO-SOURCE

BUILD SUCCESSFUL in 49s
8 actionable tasks: 8 executed
```

So, just checking to see whether it is due to the liferay-portal version of Yarn (v1.13.0) vs the one I was using when I ran `yarn build` (v1.22.5), it seems like it is:

```
$ portool yarn run build
yarn run v1.13.0
warning package.json: No license field
$ liferay-npm-scripts build
Using NODE_ENV: "production"
{ SyntaxError: /Users/greghurrell/code/portal/liferay-portal/modules/apps/commerce/commerce-product-options-web/src/main/resources/META-INF/resources/js/ProductOptions.js: Unexpected token (31:46)

  29 |  };
  30 |
> 31 |  nameInput.addEventListener('input', debounce.(handleOnNameInput, 200));
     |                                               ^
  32 |
  33 |  document
  34 |          .getElementById( namespace +'publishButton')
    at Object._raise (/Users/greghurrell/code/portal/liferay-portal/modules/node_modules/@babel/parser/lib/index.js:763:17)
  loc: Position { line: 31, column: 46 },
  pos: 1091,
  code: 'BABEL_PARSE_ERROR' }
(node:15114) UnhandledPromiseRejectionWarning: Error: Command /Users/greghurrell/code/portal/liferay-portal/modules/node_modules/@liferay/npm-scripts/node_modules/.bin/babel --no-babelrc --config-file /Users/greghurrell/code/portal/liferay-portal/modules/apps/commerce/commerce-product-options-web/build/npmscripts/tmp/.babelrc src/main/resources/META-INF/resources --out-dir build/node/packageRunBuild/resources --source-maps exited with code 1
    at spawnSync (/Users/greghurrell/code/portal/liferay-portal/modules/node_modules/@liferay/npm-scripts/src/utils/spawnSync.js:42:9)
(node:15114) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:15114) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✨  Done in 3.74s.
```

Sigh. So I created https://github.com/liferay/liferay-frontend-projects/issues/378 and will look into a fix now.

Reply:
Seems it was already fixed [as described here](https://github.com/liferay/liferay-frontend-projects/issues/378#issuecomment-753930726); you just need to rebase.